### PR TITLE
Audrey

### PR DESCRIPTION
**Pull request by AudreyKj**
fixed Captcha's color theme bug. 

- I added a key to the element where the captcha is rendered: the key changes when the color theme changes (so it will make React unmount / re-mount and we will avoid to render captcha twice). 

- a second useEffect takes care of updating captcha's color when the theme changes with a clean-up function. the only way I found to change captcha's color dynamically is through  window.location.reload() (not super satisfying i know, but it's the only solution i found)

### CODE REVIEW

**Comment by muke1908** [comment]
@AudreyKj  The key strategy looks clever, but the implementation is wrong. Let's put the key in the component state, and change it when theme is toggled.
The second useEffect block is not required. Anyway, we have to avoid refreshing page :D

File: client/src/pages/chatlink/index.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ const App = () => {
   const [captchaToken, setCaptchToken] = useState(null);
   const [loading, setLoading] = useState(false);
   const [darkMode] = useContext(ThemeContext);
+  let key = 0;
```
Comment (muke1908) [inline client/src/pages/chatlink/index.js:17]:
```suggestion
const [ elKey, setElKey ] = useState(0);
```

File: client/src/pages/chatlink/index.js [hunk 2]
```diff
@@ -42,9 +43,20 @@ const App = () => {
...
 
+  useEffect(() => {
+    let cleanUp;
+    key++;
+
+    return () => {
+      cleanUp = true;
+      window.location.reload();
+    };
+  }, [darkMode, key]);
```
Comment (muke1908) [inline client/src/pages/chatlink/index.js:58]:
```suggestion
```

File: client/src/pages/chatlink/index.js [hunk 3]
```diff
@@ -42,9 +43,20 @@ const App = () => {
 
   useEffect(() => {
     initCaptcha();
+
```
Comment (muke1908) [inline client/src/pages/chatlink/index.js:46]:
```suggestion
      setElKey(elKey+1);
      initCaptcha();
```

File: client/src/pages/chatlink/index.js [hunk 4]
```diff
@@ -71,7 +83,7 @@ const App = () => {
           </div>
           {!chatLink && (
             <>
-              <div id="captcha" className={stylesLocal.captchaHeightSetter}></div>
+              <div id="captcha" className={stylesLocal.captchaHeightSetter} key={key}></div>
```
Comment (muke1908) [inline client/src/pages/chatlink/index.js:86]:
```suggestion
              <div id="captcha" className={stylesLocal.captchaHeightSetter} key={elKey}></div>
```

**Comment by AudreyKj** [comment]
@muke1908 thanks for the review and simplifying my solution. I made the changes: 

https://github.com/muke1908/chat-e2ee/compare/master...AudreyKj:audrey

**Pull request merged**